app/components/ui/button.tsx

```diff
@@ -17,53 +17,99 @@ type BaseButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
   buttonStyle: (typeof buttonStyles)[keyof typeof buttonStyles];
   children: React.ReactNode;
+  icon?: true;
   disabled?: boolean;
 };
 
-type ButtonProps = BaseButtonProps & {
-  icon?: false;
-};
+function getVariantClasses(variant: (typeof variants)[keyof typeof variants]) {
+  switch (variant) {
+    case 'primary':
+      return 'bg-primary text-white focus:ring-gray-300 hover:bg-gray-900';
 
-type ButtonAndIconProps = BaseButtonProps & {
-  icon?: true;
-  'aria-label': string;
-};
+    case 'secondary':
+      return 'bg-secondary focus:ring-teal-300 hover:bg-teal-200';
 
-type CombinedButtonProps = ButtonProps | ButtonAndIconProps;
+    case 'warm':
+      return 'bg-warm text-white focus:ring-red-300 hover:bg-red-700';
 
-export function Button({ variant, buttonStyle, icon, disabled, children, ...props }: CombinedButtonProps) {
+    // Outlined style
+    default:
+      return 'bg-white text-primary hover:text-white border border-gray-800 hover:bg-gray-900 focus:ring-gray-300 hover:bg-gray-100';
+  }
+}
+
+function getStyleClasses(buttonStyle: (typeof buttonStyles)[keyof typeof buttonStyles]) {
+  return buttonStyle === 'rounded' ? 'rounded-lg' : 'rounded-full';
+}
+
+export function Button({ variant, buttonStyle, icon, disabled, children, ...props }: BaseButtonProps) {
   let baseClasses =
-    'w-full sm:w-fit me-2 mb-2 text-center font-medium focus:outline-none focus:ring-4 px-3 py-2 text-sm sm:px-5 sm:py-2.5 md:py-3 md:text-base';
-  let variantClasses = '';
-  let styleClasses = '';
-
-  if (variant === 'primary') {
-    variantClasses = 'bg-primary text-white focus:ring-gray-300 hover:bg-gray-900';
-  } else if (variant === 'secondary') {
-    variantClasses = 'bg-secondary focus:ring-teal-300 hover:bg-teal-200';
-  } else if (variant === 'warm') {
-    variantClasses = 'bg-warm text-white focus:ring-red-300 hover:bg-red-700';
-  } else {
-    variantClasses = 'bg-white text-primary border border-gray-200 focus:ring-gray-300 hover:bg-gray-100';
+    'w-full sm:w-fit me-2 mb-2 px-3 sm:px-5 py-2 sm:py-2.5 md:py-3 text-center text-sm md:text-base font-medium focus:outline-none focus:ring-4';
+
+  let variantClasses = getVariantClasses(variant);
+  let styleClasses = getStyleClasses(buttonStyle);
+
+  if (icon) {
+    baseClasses += ' inline-flex items-center justify-center';
   }
 
-  if (buttonStyle === 'rounded') {
-    styleClasses = 'rounded-lg';
-  } else {
-    styleClasses = 'rounded-full';
+  if (disabled) {
+    variantClasses =
+      'w-full sm:w-fit me-2 mb-2 px-3 sm:px-5 py-2 sm:py-2.5 md:py-3 text-center text-sm md:text-base font-medium bg-gray-300 text-gray-600 opacity-75 cursor-not-allowed';
+    baseClasses = '';
   }
 
+  return (
+    <button className={`${baseClasses} ${variantClasses} ${styleClasses}`} {...props}>
+      {children}
+    </button>
+  );
+}
+
+export function AuthButton({ variant, buttonStyle, icon, disabled, children, ...props }: BaseButtonProps) {
```

Comment:
The name indicates that this component does not belong in the UI folder. I suggest we really try to just use the existing `Button` component or if we really need a different design, I suggest to move this button into our auth module folder where the auth feature code is located!

Reply:
Great suggestion! I also think that we can get rid off AuthButton, which is too much for this case.